src/participant/prompts/namespace.ts

```diff
@@ -64,20 +65,22 @@ No names found.
     // message was to connect. We want to use the last
     // message they sent before the connection name as their prompt.
     let userPrompt = request.prompt;
-    if (
-      connectionNames.includes(request.prompt) &&
-      (context.history[context.history.length - 1] as vscode.ChatResponseTurn)
-        ?.result?.metadata?.askToConnect
```

Comment:
This is where the bug was - `askToConnect` was replaced with `intent`, but because this was operating on the `vscode.ChatResult.metadata`, which is of type `any`, the compiler didn't complain.